LFS-95: Resource to JSONObject AdapterFactory

zm2he opened the pull request:
In addition to the AdapterFactory, Testservlet.java has been included for the purpose of testing the AdapterFactory. It can be accessed via an authenticated GET request at `http://localhost:8080/testadapter`. This servlet should be removed prior to merging with dev.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Experimental setup for AdapterFactory

Commit:
LFS-95: Rescource to JSONObject AdapterFactory
Completed draft ResourceToJsonAdapter in resourcetojsonadapter/

Commit:
LFS-95: Adapter mored to utils module and new testing class created.

Commit:
[misc] add .metadata to gitignore

Commit:
LFS-95: Resource to JSONObject Adapterfactory
Added type handling for ValueMap object, especially for JCR Properties.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Refactor directory structure and register in provisioning

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Resolve getAdapter() type error, add handling for doubles, and add testing servlet for AdapterFactory.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Reformat date using SimpleDateFormat.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Handling for binary InputStreams, refactoring to get rid of cyclomatic complexity issues.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Refactor to remove erroneous handling of primitives.

Commit:
LFS-95: Resource to JSONObject AdapterFacotry
Add comments, rename AdapterFactory, remove testing code from dataentry.cnd.

sdumitriu commented on modules/utils/src/main/java/ca/sickkids/ccm/lfs/commons/internal/Testservlet.java:
```diff
@@ -0,0 +1,70 @@
+package main.java.ca.sickkids.ccm.lfs.commons.internal;
```
Please remove this debug class.

Commit:
LFS-95: Resource to JSONObject AdapterFactory
Add all requested changes except one regarding timezones.

Commit:
[misc] Get rid of extra whitespace.

zm2he commented on modules/utils/src/main/java/ca/sickkids/ccm/lfs/commons/internal/ResourceToJsonAdapterFactory.java:
```diff
@@ -0,0 +1,221 @@
+    {
+        GregorianCalendar calendar = (GregorianCalendar) obj;
+        // format date as year-month-day 'T' hours:minutes:seconds.milliseconds-timezone
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
```
Proper handling of this to be determined.

sdumitriu replied:
I've tested some more, and it seems that the current way is OK.

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Added a script for accessing resources using the new JSON adapter

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Cleanup: an array can never contain another array

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Better parameter names

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Better ordering of methods

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
More logical code: nested blocks should have less code, or:
branches should hold the exception from the rule, not the default case

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Shorter code, there's no need to declare a variable for a simple consumer

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Better way of casting the result, which avoids the need for a @SuppressWarnings

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Cast ASAP, methods should receive the right type of argument without the need to cast inside

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Use proper logging

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Bugfix: binary files are serialized with wrong bytes

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
More comments

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Better variable name

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Better method name

sdumitriu committed:
LFS-95: Write a generic AdapterFactory that can convert a Resource into a JSONObject
Whitespace fixes
